pkg/list/lister.go

```diff
@@ -101,27 +177,48 @@ func (l *lister) fetchRepoAndInstalled(ctx context.Context, options ListOptions)
 
 	wg.Wait()
 
-	compositeErr := multierr.Combine(repoErr, pkgErr, pkgInfoErr)
-	if pkgErr != nil || pkgInfoErr != nil {
+	compositeErr := multierr.Combine(repoErr, clPkgErr, pkgErr, pkgInfoErr)
+	if clPkgErr != nil || pkgErr != nil || pkgInfoErr != nil {
+		// repoErr is ignored here, because that way we can still return a partial list, if one of the repos is not available
 		return nil, compositeErr
 	}
 
-	result := make([]result, len(index.Packages))
+	// TODO what if a package is namespaced in one repository, and with the same name cluster scoped in another??
+
+	resultLs := make([]result, 0)
 	for i, indexPackage := range index.Packages {
-		result[i].IndexItem = &index.Packages[i]
-		for j, pkg := range packages.Items {
-			if indexPackage.Name == pkg.Name {
-				result[i].Package = &packages.Items[j]
-				packageInfoName := names.PackageInfoName(&pkg)
-				for k, packageInfo := range packageInfos.Items {
-					if packageInfo.Name == packageInfoName {
-						result[i].PackageInfo = &packageInfos.Items[k]
-						break
-					}
+		res := result{
+			IndexItem: &index.Packages[i],
+		}
+		if v1alpha1.PackageScope(indexPackage.Scope) == v1alpha1.ScopeCluster && typeOpts&includeClusterPackages != 0 {
```

Comment:
Check for empty scope is missing. 

Since this is probably going to happen a lot, I think `PackageRepoIndexItem` should use `*PackageScope` and then we can define some functions like `(*PackageScope) IsCluster() bool` and `(*PackageScope) IsNamespaced() bool` on the type directly.